autogpt_platform/backend/backend/data/block.py

```diff
@@ -492,6 +510,127 @@ def get_blocks() -> dict[str, Type[Block]]:
     return load_all_blocks()
 
 
+def is_block_provider_configured(block_cls: type["Block"]) -> bool:
+    """
+    Check if a block has a valid authentication method configured at runtime.
+
+    For example if a block is an OAuth-only block and there env vars are not set,
+    do not show it in the UI.
+
+    """
+    from backend.sdk.registry import AutoRegistry
+
+    # Create an instance to access input_schema
+    try:
+        block = block_cls.create()
+    except Exception as e:
+        # If we can't create a block instance, assume it's not OAuth-only
+        logger.error(f"Error creating block instance for {block_cls.__name__}: {e}")
+        return True
+    logger.debug(
+        f"Checking if block {block_cls.__name__} has a valid provider configured"
+    )
+
+    # Get all credentials fields from input schema
+    credentials_fields = block.input_schema.get_credentials_fields()
+    if not credentials_fields:
+        logger.debug(
+            f"Block {block_cls.__name__} has no credentials fields - Treating as valid"
+        )
+        return True
+
+    # Check required credentials fields
+    required_fields = block.input_schema.get_required_fields()
+    required_credentials = [f for f in credentials_fields if f in required_fields]
+
+    if not required_credentials:
+        logger.debug(
+            f"Block {block_cls.__name__} has no required credentials fields - Treating as valid"
+        )
+        return True
+
+    # Check if ANY required credentials field has authentication configured
```

Comment:
I think this check needs to be:
```suggestion
    # Check if ALL required credentials fields have ≥1 supported auth type configured

Reply:
Neither of our attempts here is particularly clear. I've updated it to something that is clear